File: demo/src/main/java/com/fengshuisystem/demo/entity/Package.java

package com.fengshuisystem.demo.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fengshuisystem.demo.entity.enums.Status;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.Nationalized;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.LinkedHashSet;
import java.util.Set;

@Getter
@Setter
@Entity
public class    Package {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "package_id", nullable = false)
    // id -> packageId
    private Integer id;

    @Column(name = "status")
    @Enumerated(EnumType.STRING)
    private Status status = Status.ACTIVE;

    @Size(max = 100)
    @NotNull
    @Column(name = "package_name", nullable = false, length = 100)
    private String packageName;

    @NotNull
    @Column(name = "price", nullable = false, precision = 10, scale = 2)
    private BigDecimal price;

    @NotNull
    @Lob
    @Column(name = "description", nullable = false)
    private String description;

    @NotNull
    @Column(name = "created_date", nullable = false)
    private Instant createdDate = Instant.now();

    @Size(max = 255)
    @NotNull
    @Nationalized
    @Column(name = "created_by", nullable = false)
    private String createdBy;

    @Column(name = "updated_date")
    private Instant updatedDate = Instant.now();

    @Size(max = 255)
    @Nationalized
    @Column(name = "updated_by")
    private String updatedBy;

    @OneToMany(mappedBy = "packageId", cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH, CascadeType.DETACH})
    @JsonIgnore
    private Set<ConsultationRequest> consultationRequests = new LinkedHashSet<>();

    @OneToMany(mappedBy = "packageId", cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH, CascadeType.DETACH})
    private Set<Post> posts = new LinkedHashSet<>();

    @ManyToMany(cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH, CascadeType.DETACH})
    @JoinTable(name = "package_bill",
            joinColumns = @JoinColumn(name = "package_id"),
            inverseJoinColumns = @JoinColumn(name = "bill_id"))
    private Set<Bill> bills = new LinkedHashSet<>();

}

File: src/models/Package.ts
import { Status } from './enums/Status';
import { ConsultationRequest } from './ConsultationRequest';
import { Post } from './Post';
import { Bill } from './Bill';

export class Package {
    packageId: number;
    status: Status;
    packageName: string;
    price: number;
    description: string;
    createdDate: Date;
    createdBy: string;
    updatedDate: Date | null;
    updatedBy: string | null;
    consultationRequests: Set<ConsultationRequest>;
    posts: Set<Post>;
    bills: Set<Bill>;

    constructor() {
        this.packageId = 0;
        this.status = Status.ACTIVE;
        this.packageName = '';
        this.price = 0;
        this.description = '';
        this.createdDate = new Date();
        this.createdBy = '';
        this.updatedDate = null;
        this.updatedBy = null;
        this.consultationRequests = new Set<ConsultationRequest>();
        this.posts = new Set<Post>();
        this.bills = new Set<Bill>();
    }
}
